tasks/main.yaml

```diff
@@ -23,21 +23,24 @@
     dest: /etc/containerd/config.toml
   notify: restart containerd
 
-- name: Containerd | Enable Containerd unit
-  systemd:
-    name: containerd.service
-    daemon_reload: yes
-    state: started
-    enabled: yes
-
 - name: Containerd | List containerd binaries
   find:
     paths: "{{ containerd_release_dir }}/bin"
   register: containerd_binaries
+  notify: restart containerd
 
 - name: Containerd | Symlink containerd binaries
   file:
     src: "{{ item.path }}"
     dest: "{{ bin_dir }}/{{ item.path | basename }}"
     state: link
   with_items: "{{ containerd_binaries.files }}"
+  notify: restart containerd
+
+- meta: flush_handlers
```

Comment:
why flush here ?

Reply:
Daemon-reload triggers a changed state in ansible, so it shouldn't be executing in the task, just in the handler. As I removed daemon-reload on the task that performs a state=enable+start, we need to make sure daemon-reload is executed before.